perf(linter): avoid unnecessary work in `jsx_a11y/anchor_is_valid` rule

overlookmotel opened the pull request:
Follow-on after #5223.

#5223 introduced the line `let span = jsx_el.opening_element.name.span();`. But the span is only needed when creating a diagnostic when the rule fails (cold path). Avoid the work of getting the span for the common case where the rule passes.

overlookmotel commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5341?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#5341** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5341?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* **#5340** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5340?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#5339** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5339?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @overlookmotel and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

overlookmotel commented:
I've removed the merge label as would like @Dunqing to review - these PRs are tweaks to code he wrote.

Dunqing commented:
> I've removed the merge label as would like @Dunqing to review - these PRs are tweaks to code he wrote.

I wrote a shit code 🥲

overlookmotel commented:
> I wrote a shit code 🥲

No I didn't mean it like that! It's a massive PR. The whole point of review is that 1 person will never catch everything, it needs a 2nd pair of eyes.

overlookmotel committed:
perf(linter): avoid unnecessary work in `jsx_a11y/anchor_is_valid` rule (#5341)

Follow-on after #5223.

#5223 introduced the line `let span = jsx_el.opening_element.name.span();`. But the span is only needed when creating a diagnostic when the rule fails (cold path). Avoid the work of getting the span for the common case where the rule passes.
